chore: Add setRequiredFields and setIgnoredFields documentation (#911)
SAP/cloud-sdk: docs-js/features/odata/common/operations/set-ignored-fields.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> If you want to ignore specific fields which changed since the last update of an entity, you can use the `setIgnoredFields` method.

Reviewer: [req] Technically it is not correct that the properties have not changed since the last update. Rather since the last get request in your app. Maybe just avoid it. I am still missing the purpose of "ignoring". Consider my suggestion:

```suggestion
If you changed properties on an entity, that you do not want to send in the payload of the update request, use the `setIgnoredFields` method to ignore them.
```

> In the following example, the `PATCH` request won't contain changes that were made to the `FIRST_NAME` property of the business partner.

Reviewer: [nth] Maybe it would be nice to add the step of changing the first_name property and another property to the example and stating what the payload of the request would look like.

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

If you changed properties on an entity, that you do not want to send in the payload of the update request, use the `setIgnoredFields` method to ignore them.

In the following example, the `PATCH` request won't contain changes that were made to the `FIRST_NAME` property of the business partner.

<Tabs groupId="version" defaultValue="major" values={[ { label: 'SDK 2.x', value: 'major' }, { label: 'SDK 1.x', value: 'deprecated' } ]}>
<TabItem value="major">

```ts
const { businessPartnerApi } = businessPartnerService();
// First we get a businesspartner with the id of "1".
const businessPartner = businessPartnerApi
    .requestBuilder()
    .getByKey('1')
    .execute({destinationName: 'myDestination'});

// Then we change the business partner's first name to "Steve" and their last name to "Smith".
businessPartner.then((businessPartnerResponse) => (businessPartnerResponse.firstName = 'Steve'));
businessPartner.then((businessPartnerResponse) => (businessPartnerResponse.lastName = 'Smith'));

> // Finally we update the business partner with the updated last name, ignoring the changes to their first name.

Reviewer: What do you think?

```suggestion
// Send a PATCH request with `{ "LastName" : "Smith" }`  and do not include the changed first name
```

businessPartnerApi
  .requestBuilder()
  .update(businessPartner)
  .setIgnoredFields(businessPartnerApi.schema.FIRST_NAME);
```

</TabItem>
<TabItem value="deprecated">

```ts

> // First we get a businesspartner with the id of "1".

Reviewer: ```suggestion
// Get a business partner
```

const businessPartner = BusinessPartner
    .requestBuilder()
    .getByKey('1')
    .execute({destinationName: 'myDestination'});

> // Then we change the business partner's first name to "Steve" and their last name to "Smith".

Reviewer: ```suggestion
// Change first name and last name
```

businessPartner.then((businessPartnerResponse) => (businessPartnerResponse.firstName = 'Steve'));
businessPartner.then((businessPartnerResponse) => (businessPartnerResponse.lastName = 'Smith'));

> // Finally we update the business partner with the updated last name, ignoring the changes to their first name.

Reviewer: What do you think?

```suggestion
// Send a PATCH request with `{ "LastName" : "Smith" }`  and do not include the changed first name
```
